java/maven/src/org/netbeans/modules/maven/runjar/LaunchArgPrereqsChecker.java

```diff
@@ -122,6 +123,15 @@ public boolean checkRunConfig(RunConfig config, ExecutionContext con) {
             config.setProperty(MavenExecuteUtils.RUN_APP_PARAMS, 
                     MavenExecuteUtils.joinParameters(appArgsValue));
         }
+        File workingDirectory = injectParams.getWorkingDirectory();
+        if (workingDirectory != null) {
+            config.setProperty(MavenExecuteUtils.RUN_WORKDIR,
+                    workingDirectory.getAbsolutePath());
+        }
+        Map<String, String> environmentVariables = injectParams.getEnvironmentVariables();
+        for (Map.Entry<String, String> env : environmentVariables.entrySet()) {
+            config.setProperty("Env." + env.getKey(), env.getValue());
```

Comment:
handle null `env.getValue()` ?

Reply:
The config property can not have null value in Maven :-( I didn't want to change that, this is why I've introduced `ENV_REMOVED` constant.